### PR TITLE
registry-secret replacment in overlay build

### PR DESCRIPTION
**Pull request by goern**
Signed-off-by: Christoph Görn <goern@redhat.com>

### CODE REVIEW

**Comment by goern** [comment]
This PR might be a fix for https://github.com/b4mad/racing/pull/113#issuecomment-1330341547

/assign @harshad16 
/assign @Gregory-Pereira

**Review by harshad16 — COMMENTED** [review]
This not the solution for the issue stated.

Follow these steps for resolution.
https://github.com/b4mad/racing/pull/113#issuecomment-1330794810

/hold
This PR wouldn't help in any way.

File: tasks/overlays-pipeline.yaml [hunk 1]
```diff
@@ -135,7 +135,7 @@ spec:
 
             overlays_dir = ci_conf.get("overlays_dir", "overlays")
             for overlay_env in ci_conf["overlays"]:
-
+                registry_secret = overlay_env["build"].get("registry_secret", "")
```
[the remark this answers is not in the document]
Comment (harshad16) [reply tasks/overlays-pipeline.yaml]: everything on that file needs to be reverted
there is no value add.
Remove this whole commit https://github.com/AICoE/aicoe-ci/pull/187/commits/dec453ce5b1ca88dfde58fb59e14dbc59b2b25f1

[the remark this answers is not in the document]
Comment (goern) [reply tasks/overlays-pipeline.yaml]: How is the secret passed on?

[the remark this answers is not in the document]
Comment (harshad16) [reply tasks/overlays-pipeline.yaml]: That tasks only run in the tag release pipeline.
As the error, we are observing is in the pull request pipeline.

the pr-configuration tasks, captures the registry-secret:
https://github.com/AICoE/aicoe-ci/blob/516849c28ff131fed8ac96c51dc2e5e557529ed3/tasks/pr-config.yaml#L122.
and pass it to the pr-build tasks.
https://github.com/AICoE/aicoe-ci/blob/516849c28ff131fed8ac96c51dc2e5e557529ed3/pipeline/pull-request-pipeline.yaml#L410

so all is already set.

only issue was that , you were trying /deploy on overlays repo, which is not designed to be called out per overlay.
some additional refactoring is needed to let user know that command is not correct.
will open an issue for this.

**Comment by goern** [comment]
/close 
this in favor of #190 and #189

**Pull request closed without merging**